deepchem/molnet/load_function/uspto_datasets.py

```diff
@@ -1,96 +1,122 @@
+    super(_USPTOLoader, self).__init__(*args, **kwargs)
+    self.subset = subset
+    self.sep_reagent = sep_reagent
+    self.name = 'USPTO_' + subset
+
+  def create_dataset(self) -> Dataset:
+    #####INCOMPLETE/INCORRECT: I don't think this is the right way to bypass the splitter!
+    if self.subset not in ['MIT', 'STEREO']:
+      raise ValueError("Valid Subset names are MIT and STEREO.")
+
+    if self.subset == 'MIT':
+      train_file = os.path.join(self.data_dir, USPTO_MIT_TRAIN)
+      test_file = os.path.join(self.data_dir, USPTO_MIT_TEST)
+      valid_file = os.path.join(self.data_dir, USPTO_MIT_VALID)
+
+      if not os.path.exists(train_file):
+        logger.info("Downloading training file...")
+        dc.utils.data_utils.download_url(
+            url=USPTO_MIT_TRAIN, dest_dir=self.data_dir)
+        logger.info("Training file download complete.")
+
+        logger.info("Downloading test file...")
+        dc.utils.data_utils.download_url(
+            url=USPTO_MIT_TEST, dest_dir=self.data_dir)
+        logger.info("Test file download complete.")
+
+        logger.info("Downloading validation file...")
+        dc.utils.data_utils.download_url(
+            url=USPTO_MIT_VALID, dest_dir=self.data_dir)
+        logger.info("Validation file download complete.")
+
+    if self.subset == 'STEREO':
+      train_file = os.path.join(self.data_dir, USPTO_STEREO_TRAIN)
+      test_file = os.path.join(self.data_dir, USPTO_STEREO_TEST)
+      valid_file = os.path.join(self.data_dir, USPTO_STEREO_VALID)
+
+      if not os.path.exists(train_file):
+        logger.info("Downloading training file...")
+        dc.utils.data_utils.download_url(
+            url=USPTO_STEREO_TRAIN, dest_dir=self.data_dir)
+        logger.info("Training file download complete.")
+
+        logger.info("Downloading test file...")
+        dc.utils.data_utils.download_url(
+            url=USPTO_STEREO_TEST, dest_dir=self.data_dir)
+        logger.info("Test file download complete.")
+
+        logger.info("Downloading validation file...")
+        dc.utils.data_utils.download_url(
+            url=USPTO_STEREO_VALID, dest_dir=self.data_dir)
+        logger.info("Validation file download complete.")
+
+    loader = dc.data.CSVLoader(
+        tasks=self.tasks, feature_field="smiles", featurizer=self.featurizer)
+
+    logger.info("Loading train dataset..")
+    train_file = loader.create_dataset(train_file, shard_size=8192)
+    logger.info("Loading test dataset..")
+    test_file = loader.create_dataset(test_file, shard_size=8192)
+    logger.info("Loading validation dataset..")
+    valid_file = loader.create_dataset(valid_file, shard_size=8192)
+    logger.info("Loading successful!")
+
+    #need to figure out how to return the train, test and valid files!
+    return (train_file, test_file, valid_file)
+
+    def load_precomputed_splits(self) -> List[Dataset, ...]:
+      pass
+
+
+def load_uspto(
```

Comment:
This function will need a docstring.

Reply:
yup! I will update it once I am done with the other methods.